WordPress/src/main/java/org/wordpress/android/datasets/AsyncTaskHandler.kt

```diff
@@ -0,0 +1,33 @@
+package org.wordpress.android.datasets
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+/**
+ * Helper class to handle async tasks by using coroutines
+ * @see <a href="https://github.com/wordpress-mobile/WordPress-Android/pull/20937">Introduction</a>
+ */
+object AsyncTaskHandler {
+    /**
+     * Load data in the background and handle the result on the main thread
+     */
+    @JvmStatic
+    fun <T> load(backgroundTask: () -> T, callback: AsyncTaskCallback<T>) {
+        CoroutineScope(Dispatchers.IO).launch {
```

Comment:
I have a few concerns about the usage of Coroutines here.

### CoroutineScope
The `CoroutineScope` is being created ad hoc instead of using an existing Scope or Context/Job, which is usually attached to some lifecycle-aware component. This is essentially the same as using the `GlobalScope`, which [should be avoided](https://developer.android.com/kotlin/coroutines/coroutines-best-practices#global-scope).

This means that the `Job` launched here is completely standalone, bypassing one of the biggest advantages introduced by coroutines: structured concurrency.

In practice, this means that even if the caller of the coroutine dies, the callback lambda will still be called (since the Scope is never canceled) which can cause issues if said callback references something that was destroyed.

TBH I'm not sure what would be the best solution here, but using an existing Scope that's close to the component that launches this Job would be preferred, all-in-all, thinking about this helper class, I'd say receiving the `CoroutineScope` via argument would make more sense so the caller can pass an appropriate Scope.

### Hardcoded Dispatchers.IO
This is another [thing that should be avoided](https://developer.android.com/kotlin/coroutines/coroutines-best-practices#inject-dispatchers), not only because it's hard to test, but also because, based on the name of this class (async **TASK**), it looks like this is supposed to be a generic "offload-to-background" helper.

This means that passing a `backgroundTask` that runs computational work could be expected, but that kind of work should be run in the `Dispatchers.Default`, while `Dispatchers.IO` should be used only for IO/waiting tasks. [More info here](https://developer.android.com/kotlin/coroutines/coroutines-adv#main-safety).

The solution for this should probably receive the `CoroutineDispatcher` as an argument since the caller would be the one who knows what kind of task is being run.

Reply:
@thomashorta Thank you for pointing out the potential issues in the current implementation.

> I'd say receiving the CoroutineScope via argument would make more sense so the caller can pass an appropriate Scope.

Yeah this is the way to handle the lifecycle stuff properly, will update it later.

Reply:
For the Hardcoded Dispatchers.IO, I'm not sure if I should receive the CoroutineDispatcher as an argument since the initial idea was to address issues related to accessing the database, opening it up to all dispatchers seems a bit over-engineered to me. 

Therefore, I propose we keep this part as is and handle it through naming for now. If a future need arises, we can then add the dispatcher argument. What do you think?